lib/logstash/filters/kv.rb

```diff
@@ -158,6 +158,20 @@ class LogStash::Filters::KV < LogStash::Filters::Base
   #     }
   config :default_keys, :validate => :hash, :default => {}
 
+  # A bool option for removing duplicate key/value pairs. When set to true, only 
+  # one key/value pair will exist, otherwise both will.
+  #
+  # For example, consider a source like `from=me from=me`. `[from]` will map to 
+  # an Array with two elements: `["me", "me"]`. to only keep unique key/value pairs,
+  # you could use this configuration:
+  # [source,ruby]
+  #     filter {
+  #       kv {
+  #         allow_duplicate_values => true
+  #       }
+  #     }
+  config :allow_duplicate_values, :validate => :boolean, :default => false
```

Comment:
some may passing arrays in url get params, this behavior would reduce that array into a set thereby changing the meaning of the parameter.